views/course_form/views.py

```diff
@@ -19,10 +19,9 @@ def get(self, request, code: str | None = None, semester: str | None = None):
         if not self.__can_use_form(request.user, code, semester):
             return redirect(reverse("home"))
 
-        course_form = CourseController.get_course(code, semester) if code else {}
-
         return render(request, 'course_form/course_form.html', {
-            "data": course_form,
+            "existSemester": SemesterController.list_semester(),
+            "data": CourseForm,
```

Comment:
missing course_form actual data. Unsure if it was a bad merge with main or if you haven't merged with main yet.

Reply:
course_form looks like it is behind